examples/dmd/wave_equation.cpp

```diff
@@ -0,0 +1,621 @@
+   //    natural.
+   FunctionCoefficient u_0(InitialSolution);
+   u_gf.ProjectCoefficient(u_0);
+   Vector u;
+   u_gf.GetTrueDofs(u);
+
+   FunctionCoefficient dudt_0(InitialRate);
+   dudt_gf.ProjectCoefficient(dudt_0);
+   Vector dudt;
+   dudt_gf.GetTrueDofs(dudt);
+
+   // 7. Initialize the conduction operator and the visualization.
+   Array<int> ess_bdr;
+   if (mesh->bdr_attributes.Size())
+   {
+      ess_bdr.SetSize(mesh->bdr_attributes.Max());
+
+      if (dirichlet)
+      {
+         ess_bdr = 1;
+      }
+      else
+      {
+         ess_bdr = 0;
+      }
+   }
+
+   WaveOperator oper(fespace, ess_bdr, speed);
+
+   u_gf.SetFromTrueDofs(u);
+   {
+      ofstream omesh("wave_equation.mesh");
+      omesh.precision(precision);
+      mesh->Print(omesh);
+      ofstream osol("wave_equation-init.gf");
+      osol.precision(precision);
+      u_gf.Save(osol);
+      dudt_gf.Save(osol);
+   }
+
+   VisItDataCollection visit_dc("Wave_Equation", mesh);
+   visit_dc.RegisterField("solution", &u_gf);
+   visit_dc.RegisterField("rate", &dudt_gf);
+   if (visit)
+   {
+      visit_dc.SetCycle(0);
+      visit_dc.SetTime(0.0);
+      visit_dc.Save();
+   }
+
+   socketstream sout;
+   if (visualization)
+   {
+      char vishost[] = "localhost";
+      int  visport   = 19916;
+      sout.open(vishost, visport);
+      if (!sout)
+      {
+         cout << "Unable to connect to GLVis server at "
+              << vishost << ':' << visport << endl;
+         visualization = false;
+         cout << "GLVis visualization disabled.\n";
+      }
+      else
+      {
+         sout.precision(precision);
+         sout << "solution\n" << *mesh << dudt_gf;
+         sout << flush;
+      }
+   }
+
+   StopWatch fom_timer, dmd_training_timer, dmd_prediction_timer;
+   fom_timer.Start();
+
+   // 8. Perform time-integration (looping over the time iterations, ti, with a
+   //    time-step dt).
+   ode_solver->Init(oper);
+   double t = 0.0;
+
+   vector<double> ts;
+
+   fom_timer.Stop();
+
+   dmd_training_timer.Start();
+   
+   int curr_window = 0;
+   vector<CAROM::DMD*> dmd_u;
+   dmd_u.push_back(new CAROM::DMD(u_gf.GetTrueVector().Size(), dt));
+   dmd_u[curr_window]->takeSample(u_gf.GetTrueVector(), t);
+   ts.push_back(t);
+
+
+   dmd_training_timer.Stop();
+
+   u_gf.SetFromTrueDofs(u);
+
+
+   bool last_step = false;
+   for (int ti = 1; !last_step; ti++)
+   {
+      fom_timer.Start();
+      if (t + dt >= t_final - dt/2)
+      {
+         last_step = true;
+      }
+
+
+      ode_solver->Step(u, dudt, t, dt);
+      fom_timer.Stop();
+
+      dmd_training_timer.Start();
+      u_gf.SetFromTrueDofs(u);
+      dmd_u[curr_window]->takeSample(u_gf.GetTrueVector(), t);
+
+      if (last_step || (ti % windowNumSamples) == 0)
+      {
+         cout << "step " << ti << ", t = " << t << endl;
+
+         if(rdim != -1){
+            
+            if(myid == 0){
+
+               std::cout << "Creating DMD with rdim: " << rdim << std::endl;
```

Comment:
`std::cout` -> `cout` throughout the file

Reply:
Fixed.